test/gpu_tests.jl

```diff
@@ -0,0 +1,31 @@
+using StaticArrays, CUDA
+
+CUDA.allowscalar(false)
+
+N = 2
+
+function rosenbrock(x, p)
```

Comment:
Start with a simpler problem. Just a quadratic https://github.com/SciML/Optimization.jl/pull/619/files#diff-4acd7752f13fc9b546d19e7d51fe086442ffdc68bef35cdbbfb2f565060cdb92R7-R18 to ensure correctness, it should converge in 1 or 2 iterations